### PR TITLE
feat: identify type of log

### PR DESCRIPTION
**Pull request by v1v**
Enable `event.dataset` to identify what type of log.
Add Workflow URL to help with the reporting.

### CODE REVIEW

File: receiver.go [hunk 1]
```diff
@@ -182,6 +182,8 @@ func (rec *githubactionsannotationsreceiver) processAnnotations(ctx context.Cont
 	resourceAttributes := resourceLogs.Resource().Attributes()
 	serviceName := generateServiceName(rec.config, repository.FullName)
 	resourceAttributes.PutStr("service.name", serviceName)
+	resourceAttributes.PutStr("github.event", "github-actions")
```
Comment (reakaleek) [inline receiver.go]: IMO, we should use something more custom.

Using the existing event attribute may confuse us in the future, as this GitHub event doesn't exist.

Comment (v1v) [reply receiver.go]: what do you suggest?

Comment (v1v) [reply receiver.go]: maybe `github.log`:
- `github.log: annotations`
- `github.log: github-actions`

then eventually:
- `github.log: audit`


-------

Although I see the concept of `event.dataset`, https://github.com/elastic/integrations/tree/main/packages/github uses 
- `"dataset": "github.secret_scanning"`
- `"dataset": "github.dependabot"`
- `"dataset": "github. code_scanning"`
- `"dataset": "github.audit",`


We could use
- `event.dataset: github.annotations`
- `event.dataset: github.actions`

Comment (reakaleek) [reply receiver.go]: I like it.

I was also looking into this and I found this
https://opentelemetry.io/docs/specs/otel/logs/data-model/#field-resource

Looks like the same concept.

Comment (v1v) [reply receiver.go]: Ok, do we agree to use the `Resource` attributes for the logs with `event.dataset` and the values ('github.annotations', 'github.actions')?

Comment (reakaleek) [reply receiver.go]: Uh oh, I think I confused something. I somehow thought there would be a field called `resource` in elasticsearch. but we are already using resource attributes 😅 

But yes, event.dataset sounds good to me.

**Review by reakaleek — APPROVED** [review]

**Pull request merged**